app/controllers/api/accounts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Api::AccountsController < ApplicationController
+  def new
+    @user = current_user
+  end
+
+  def create
+    @user = current_user
+    @user_api = UserApi.find_or_create_by(user_id: @user.id)
+    if valid_password? && params[:accept]
```

Comment:
That should eliminate the need for one of these `if` statements. Then from the view you can tell it where to go instead of asking here in the controller.